### PR TITLE
Additional RTTI Fix

### PR DESCRIPTION
**Pull request by vraghavulu**
Fix RTTI declaration

Details:
 - issues in Android system with the standart c++ RTTI changing type cast of std::dynamic_pointer_cast to ov::as_type_ptr

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Comment by t-jankowski** [comment]
@mitruska Look on changes at src/core/include/openvino/op

File: src/core/include/openvino/pass/pattern/op/pattern.hpp [hunk 1]
```diff
@@ -83,6 +83,7 @@ ValuePredicate as_value_predicate(NodePredicate pred);
 
 class OPENVINO_API Pattern : public Node {
 public:
+    OPENVINO_RTTI("pattern");
```
Comment (t-jankowski) [inline src/core/include/openvino/pass/pattern/op/pattern.hpp:86]:
```suggestion
    OPENVINO_RTTI("Pattern");
```

File: src/core/include/openvino/op/adaptive_avg_pool.hpp [hunk 2]
```diff
@@ -15,7 +15,7 @@ namespace v8 {
 /// \ingroup ov_ops_cpp_api
 class OPENVINO_API AdaptiveAvgPool : public Op {
 public:
-    OPENVINO_OP("AdaptiveAvgPool", "opset8");
+    OPENVINO_OP("AdaptiveAvgPool", "opset8", Op);
```
Comment (mitruska) [inline src/core/include/openvino/op/adaptive_avg_pool.hpp:18]: Does the fix require adding the "Op" to each `OPENVINO_OP` macro? 
The `ov::op::Op` is the default for `OPENVINO_OP` macro defined here:
https://github.com/openvinotoolkit/openvino/blob/571e98d5880a30e9d8ca25f445c343e955e79123/src/core/include/openvino/op/op.hpp#L14-L21

**Comment by t-jankowski** [comment]
To solve issue with failing 'Code Style / clang-format' it should be enough (it works on my side) to replace sub-string `,"0",` with `, "0",` in the changes of this PR.
Note there are build targets `clang_format_check_all` and `clang_format_fix_all` which help with code style issues.

**Comment by vraghavulu** [comment]
> To fix failing `ov_core_unit_tests` `ov_transformations_tests` (which affect several CI jobs) RTTI definitions in below files need to be updated src/common/transformations/include/transformations/common_optimizations/glu_fusion.hpp src/common/transformations/include/transformations/common_optimizations/lora_subgraph_fusion.hpp src/common/transformations/include/transformations/fp16_compression/mark_floatpoint_range.hpp src/common/transformations/include/transformations/op_conversions/convert_slicescatter.hpp src/common/transformations/include/transformations/op_conversions/convert_squeeze15_downgrade.hpp src/core/tests/graph_rewrite.cpp src/core/tests/pass_config.cpp
> 
> I've noticed there are more not changed this way RTTI definitions in src/common/transformations, but these don't affect the tests on my side. However I think these should also be updated. `grep -Erl 'OPENVINO_RTTI.+"\s*);' src/common/transformations/`



In the file src/common/transformations/include/transformations/op_conversions/convert_squeeze15_downgrade.hpp, the parent class is "MatcherPass" only. Should the RTTI definition for parent class be given as "MatcherPass" or "ov::pass::MatcherPass"

**Comment by t-jankowski** [comment]
> > To fix failing `ov_core_unit_tests` `ov_transformations_tests` (which affect several CI jobs) RTTI definitions in below files need to be updated src/common/transformations/include/transformations/common_optimizations/glu_fusion.hpp src/common/transformations/include/transformations/common_optimizations/lora_subgraph_fusion.hpp src/common/transformations/include/transformations/fp16_compression/mark_floatpoint_range.hpp src/common/transformations/include/transformations/op_conversions/convert_slicescatter.hpp src/common/transformations/include/transformations/op_conversions/convert_squeeze15_downgrade.hpp src/core/tests/graph_rewrite.cpp src/core/tests/pass_config.cpp
> > I've noticed there are more not changed this way RTTI definitions in src/common/transformations, but these don't affect the tests on my side. However I think these should also be updated. `grep -Erl 'OPENVINO_RTTI.+"\s*);' src/common/transformations/`
> 
> In the file src/common/transformations/include/transformations/op_conversions/convert_squeeze15_downgrade.hpp, the parent class is "MatcherPass" only. Should the RTTI definition for parent class be given as "MatcherPass" or "ov::pass::MatcherPass"

Shorter version is preferred. In this case it's in `namespace ov::pass` scope and both point the same definition.

File: src/core/include/openvino/pass/pattern/op/pattern.hpp (hunk 1 above)
Comment (t-jankowski) [reply src/core/include/openvino/pass/pattern/op/pattern.hpp]: Done

File: src/core/include/openvino/op/adaptive_avg_pool.hpp (hunk 2 above)
Comment (t-jankowski) [reply src/core/include/openvino/op/adaptive_avg_pool.hpp]: It's not required - reverted.

File: src/common/transformations/include/transformations/common_optimizations/gelu_fusion.hpp [hunk 3]
```diff
@@ -64,7 +64,7 @@ class ov::pass::GeluFusionWithErfThree : public ov::pass::MatcherPass {
  */
 class ov::pass::GeluFusionWithErfFour : public ov::pass::MatcherPass {
 public:
-    OPENVINO_RTTI("GeluFusionWithErfFour", "0");
+    OPENVINO_RTTI("GeluFusionWithErfFour", "0", ov::pass::MatcherPass);
```
Comment (ilya-lavrenov) [inline src/common/transformations/include/transformations/common_optimizations/gelu_fusion.hpp:67]: maybe we need to introduce `OPENVINO_MATCHER_PASS_RTTI` ? as a shortcut

**Comment by t-jankowski** [comment]
build_jenkins

**Comment by t-jankowski** [comment]
Due to to review complexity it's been decided to do these changes in bunch of small targeted PRs - tracked under epic CVS-159217

**Pull request closed without merging**